src/modules/rc_loss_alarm/rc_loss_alarm.cpp

```diff
@@ -0,0 +1,235 @@
+int RC_Loss_Alarm::task_spawn(int argc, char *argv[])
+{
+	// Instantiate task
+	RC_Loss_Alarm *rc_loss_alarm = nullptr;
+
+	if (_object == nullptr) {
+		rc_loss_alarm = new RC_Loss_Alarm();
+
+		if (rc_loss_alarm == nullptr) {
+			PX4_ERR("failed to start flow");
+			return PX4_ERROR;
+		}
+	}
+
+	_object = rc_loss_alarm;
+
+	// schedule a cycle to start things
+	int ret = work_queue(LPWORK, &_work,
+			     (worker_t)&RC_Loss_Alarm::cycle_trampoline, rc_loss_alarm, 0);
+
+	if (ret < 0) {
+		return ret;
+	}
+
+	// Since this task runs on the work-queue rather than in a separate thread,
+	// we specify the workqueue task_id
+	_task_id = task_id_is_work_queue;
+
+	// wait until task is up & running (the mode_* commands depend on it)
+	if (wait_until_running() < 0) {
+		return -1;
+	}
+
+	return PX4_OK;
+}
+
+/** @see ModuleBase */
+int RC_Loss_Alarm::custom_command(int argc, char *argv[])
+{
+	if (!strcmp(argv[0], "test")) {
+		RC_Loss_Alarm::pub_tune();
+		return PX4_OK;
+
+	} else if (!strcmp(argv[0], "reset")) {
+		return RC_Loss_Alarm::reset_module();
+	}
+
+	return print_usage("unknown command");
+}
+
+/** @see ModuleBase */
+int RC_Loss_Alarm::print_usage(const char *reason)
+{
+	if (reason) {
+		PX4_WARN("%s\n", reason);
+	}
+
+	PRINT_MODULE_DESCRIPTION(
+		R"DESCR_STR(
+### Description
+RC Loss Alarm plays a loud error tune in the event that the drone loses RC
+link after disarming. Ideal for finding a lost drone. The RC Loss Module
+subscribes to the vehicle_status topic and checks the armed state as well as
+the rc_signal_lost flag. The RC Loss alarm will only be triggered if
+rc_signal_lost flag was false at least once at some point (i.e. if an RC is
+connected).
+### Example
+The module is typically started with:
+rc_loss_alarm start
+)DESCR_STR");
+
+	PRINT_MODULE_USAGE_NAME("rc_loss_alarm", "driver");
+	PRINT_MODULE_USAGE_COMMAND_DESCR("start",  "Start the task");
+	PRINT_MODULE_USAGE_COMMAND_DESCR("stop",   "Stop the task");
+	PRINT_MODULE_USAGE_COMMAND_DESCR("status", "Display running status of task");
+	PRINT_MODULE_USAGE_COMMAND_DESCR("test",   "Start alarm tune, can be stopped with 'reset'");
+	PRINT_MODULE_USAGE_COMMAND_DESCR("reset",  "Resets internal state and stops an eventual alarm");
+	return PX4_OK;
+}
+
+void RC_Loss_Alarm::cycle_trampoline(void *arg)
+{
+	RC_Loss_Alarm *dev = reinterpret_cast<RC_Loss_Alarm *>(arg);
+	dev->cycle();
+}
+
+void RC_Loss_Alarm::cycle()
+{
+	// Subscribe if necessary
+	if(_vehicle_status_sub == -1){
+		_vehicle_status_sub = orb_subscribe(ORB_ID(vehicle_status));
+	}
+
+	// Check armed status
+	bool updated = false;
+
+	orb_check(_vehicle_status_sub, &updated);
+	if(updated){
+		orb_copy(ORB_ID(vehicle_status), _vehicle_status_sub, &_vehicle_status);
+
+		if (!_was_armed &&
+				_vehicle_status.arming_state == vehicle_status_s::ARMING_STATE_ARMED){
+
+			_was_armed = true;	// Once true, impossible to go back to false
+		}
+
+		if (!_had_rc && !_vehicle_status.rc_signal_lost){
```

Comment:
I think you'll want to use rc_signal_found_once as well. It's currently in vehicle_status_flags.

Reply:
:heavy_check_mark: